archive/rule.go

```diff
@@ -1,34 +1,53 @@
 package archive
 
 import (
-	"errors"
 	"fmt"
 	"path/filepath"
 	"strings"
 
 	"github.com/brimsec/zq/ast"
 	"github.com/brimsec/zq/zng/resolver"
+	"github.com/brimsec/zq/zqe"
 	"github.com/brimsec/zq/zql"
 )
 
-func ParsePattern(in string) (string, string, error) {
+type IndexQuery struct {
```

Comment:
nittish: do IndexName and Patterns need to be exported?

Reply:
they definitely don't right now, so I'll un-export them.